### PR TITLE
Feature/create route map

### PR DESCRIPTION
**Pull request by LennartSchwier**
Add functionality to allow the user to create a new route by placing the waypoints on a map

### CODE REVIEW

File: frontend/public/index.html [hunk 1]
```diff
@@ -2,16 +2,16 @@
 <html lang="en">
   <head>
     <meta charset="utf-8" />
-    <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
+    <link rel="icon" href="%PUBLIC_URL%/saildog-logo.png" />
```
Comment (jamarob) [inline frontend/public/index.html:5]: You should keep the name `favicon.ico` (and the logo variants). They are referenced in the manifest file which is important for turning your webapp into a PWA.

File: frontend/src/components/route/NewRouteMap.js [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+                    <Polyline path={waypoints}/>
+                </GoogleMap>
+                <button className={"cancel"} onClick={redirectBackToRoutes}><MdCancel/>Cancel</button>
+                <button className={"done"} onClick={createRoute}><MdDone/>Done</button>
+            </MapContainer>
+        );
+    }
+
+    function centerMap() {
+        if (waypoints.length === 0) {
```
Comment (jamarob) [inline frontend/src/components/route/NewRouteMap.js:77]: Maybe turn it around and check for a truthy value of `waypoint.length`?

File: frontend/src/components/route/NewRouteMap.js [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+    }
+
+    function labelMarker(collection, item) {
+        if ((collection.indexOf(item)) === 0) {
+            return "Start";
+        }
+        if ((collection.indexOf(item) + 1) === collection.length) {
+            return "End";
+        }
+        return String(collection.indexOf(item));
```
Comment (jamarob) [inline frontend/src/components/route/NewRouteMap.js:100]: A more ideomatic way to turn it into a string is `""+`. Better find the index at the beginning of the function and save it into a variable, saves you two iterations over the array.

File: frontend/src/components/route/NewRouteMap.js [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+    }
+
+    function saveRouteName() {
```
Comment (jamarob) [inline frontend/src/components/route/NewRouteMap.js:103]: Maybe rename to `toggleShowName`? Just from reading the function name I expect it to persistet the route name.

File: frontend/src/components/route/NewRouteMap.js [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+            }
+            if (i > 0) {
+                 legs[i] = {
+                    startLatitude: waypointsWithoutStart[i-1].lat.toFixed(6),
+                    startLongitude: waypointsWithoutStart[i-1].lng.toFixed(6),
+                    endLatitude: waypointsWithoutStart[i].lat.toFixed(6),
+                    endLongitude: waypointsWithoutStart[i].lng.toFixed(6),
+                }
+            }
+        }
```
Comment (jamarob) [inline frontend/src/components/route/NewRouteMap.js:136]:
```suggestion
        const legs = waypointsWithoutStart.map((waypoint ,index)=> {
            if(index === 0){
                return {
                    startLatitude: startWaypoint.startLatitude.toFixed(6),
                    startLongitude: startWaypoint.startLongitude.toFixed(6),
                    endLatitude: waypointsWithoutStart[i].lat.toFixed(6),
                    endLongitude: waypointsWithoutStart[i].lng.toFixed(6),
                };
            }
            return  {
                    startLatitude: waypointsWithoutStart[i-1].lat.toFixed(6),
                    startLongitude: waypointsWithoutStart[i-1].lng.toFixed(6),
                    endLatitude: waypointsWithoutStart[i].lat.toFixed(6),
                    endLongitude: waypointsWithoutStart[i].lng.toFixed(6),
                }
            };
        });
```
functional style > imperative style :)

File: frontend/src/components/route/NewRouteMap.js [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+  .saveName {
+  top: 1rem;
+  right: 1rem;
+  }
+  
+  .cancel, .done {
+  display: grid;
+  justify-items: center;
+  bottom: 2rem;
+  }
```
Comment (jamarob) [inline frontend/src/components/route/NewRouteMap.js:177]: Can be added to `button`

File: frontend/src/components/route/RouteDetails.js [hunk 7]
```diff
@@ -29,10 +29,12 @@ export default function RouteDetails() {
             <PageLayout>
                 <Header headerText={route?.routeName}/>
                 <ul>
+                    <div className={"placeholder"}/>
                     {!legs? null :
                         legs.map((leg, index) => <Leg key={leg.legId} leg={leg} index={index}/>)
                     }
                     <RouteEnd endWaypoint={endWaypoint}/>
+                    <div className={"placeholder"}/>
```
Comment (jamarob) [inline frontend/src/components/route/RouteDetails.js:37]: Semantically only `li` are allowed in `ul`. if you need space at the bottom, select the last child and add a `::after` pseudo element ;)

**Review by jamarob — COMMENTED** [review]
Good job! ⛵

File: frontend/public/index.html (hunk 1 above)
Comment (LennartSchwier) [reply frontend/public/index.html]: I have now added code that was automatically generated. Hope that is better.

**Comment by LennartSchwier** [comment]
@jamarob thanks a lot for your review and all the suggestions, which I happily implemented. 😊
Unfortunately I merged the branch were I implemented prettier into this one before checking for any comments/suggestions. That’s why all of your contribution was outdated.

File: frontend/public/index.html (hunk 1 above)
Comment (jamarob) [reply frontend/public/index.html]: Just a precaution to save you some headache if you decide to turn it into a PWA ;)

**Pull request merged**